src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -86,153 +90,139 @@ public FunctionStatus GetFunctionStatus(string name)
-                    {
-                        ["Source"] = logEntry.Source,
-                        ["FunctionName"] = logEntry.FunctionName
-                    };
-                    _logger.Log(logLevel, 0, logData, null, (s, e) => logEntry.Message);
+                    traceEvent.Properties.Add(ScriptConstants.TracePropertyFunctionNameKey, logEntry.FunctionName);
                 }
 
-                return Ok();
-            }
-            else
-            {
-                return Unauthorized();
+                var logLevel = Utility.ToLogLevel(traceEvent.Level);
+                var logData = new Dictionary<string, object>
+                {
+                    ["Source"] = logEntry.Source,
+                    ["FunctionName"] = logEntry.FunctionName
+                };
+                _logger.Log(logLevel, 0, logData, null, (s, e) => logEntry.Message);
             }
+
+            return Ok();
         }
 
         [HttpPost]
         [Route("admin/host/debug")]
-        public HttpResponseMessage LaunchDebugger()
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public IActionResult LaunchDebugger()
         {
             if (_webHostSettings.IsSelfHost)
             {
                 // If debugger is already running, this will be a no-op returning true.
                 if (Debugger.Launch())
                 {
-                    return new HttpResponseMessage(HttpStatusCode.OK);
+                    return Ok();
                 }
                 else
                 {
-                    return new HttpResponseMessage(HttpStatusCode.Conflict);
+                    return StatusCode(StatusCodes.Status409Conflict);
                 }
             }
-            return new HttpResponseMessage(HttpStatusCode.NotImplemented);
+
+            return StatusCode(StatusCodes.Status501NotImplemented);
         }
 
-        public override Task<HttpResponseMessage> ExecuteAsync(HttpControllerContext controllerContext, CancellationToken cancellationToken)
+        public override void OnActionExecuting(ActionExecutingContext context)
         {
             // For all admin api requests, we'll update the ScriptHost debug timeout
             // For now, we'll enable debug mode on ANY admin requests. Since the Portal interacts through
             // the admin API this is sufficient for identifying when the Portal is connected.
             _scriptHostManager.Instance?.NotifyDebug();
 
-            return base.ExecuteAsync(controllerContext, cancellationToken);
+            base.OnActionExecuting(context);
         }
 
-        [Route("admin/extensions/{name}/{*extra}")]
         [HttpGet]
         [HttpPost]
-        [AllowAnonymous]
-        public async Task<HttpResponseMessage> ExtensionWebHookHandler(string name, CancellationToken token)
+        [Authorize(AuthenticationSchemes = AuthLevelAuthenticationDefaults.AuthenticationScheme)]
+        [Route("webhookextensions/handler/{name}/{*extra}")]
```

Comment:
The "handler" portion of this route doesn't feel right. If "name" is indexing into handlers, shouldn't it be `webhookextensions/handers/{name}/{*extra}`?

Reply:
You're right, this doesn't look correct. I'll go with your 2nd option.

Reply:
How about just `webhookextensions/{name}/{*extra}` ? We could also change `webhookextensions` to something like `functionswebhookextensions` to make it more specific to functions